src/box/field_def.h

```diff
@@ -182,8 +195,16 @@ field_mp_type_is_compatible(enum field_type type, const char *data,
 	assert((size_t)mp_type < CHAR_BIT * sizeof(*field_mp_type));
 	uint32_t mask;
 	if (mp_type != MP_EXT) {
-		return field_mp_plain_type_is_compatible(type, mp_type,
-							 is_nullable);
+		if (likely(field_mp_plain_type_is_compatible(type, mp_type,
```

Comment:
Seem to be a logical mistake. If plain type is compatible then general type which is plain must be compatible too.

Reply:
Implementation has been changed.